SimPEG/potential_fields/magnetics/simulation.py

```diff
@@ -283,33 +280,33 @@ def evaluate_integral(self, receiver_location, components):
             #     # inside an active cell.
             #     node_evals["gzz"] = -node_evals["gxx"] - node_evals["gyy"]
 
-        if "bxx" in components:
+        if "bxx" in components or "tmi_x" in components:
             node_evals["gxxx"] = prism_fzzz(dy, dz, dx)
             node_evals["gxxy"] = prism_fxxy(dx, dy, dz)
             node_evals["gxxz"] = prism_fxxz(dx, dy, dz)
-        if "bxy" in components:
+        if any(s in components for s in ["bxy", "tmi_x", "tmi_y"]):
```

Comment:
I know this is a mouthful, but it's more efficient than having to build a list and a tuple on each iteration and then checking if any of its values is True.

What do you think?

```suggestion
        if "bxy" in components or "tmi_x" in components or "tmi_y" in components:
```

Reply:
even cleaner would be to do a set().intersection

```suggestion
        if any(set(["bxy", "tmi_x", "tmi_y"]).intersection(components)):
```

Reply:
@santisoler : you are right, your solution is faster.
Cool syntax @domfournier , noted. But I will go with Santi's suggestion.
<img width="693" alt="image" src="https://github.com/simpeg/simpeg/assets/97514898/8f44cbcd-7284-4520-bec3-674b6800a5ec">